target_snowflake/__init__.py

```diff
@@ -303,7 +306,7 @@ def flush_streams(
     if filter_streams:
         streams_to_flush = {k: v for k, v in streams.items() if k in filter_streams}
     else:
-        streams_to_flush = streams
+        streams_to_flush = copy.deepcopy(streams)
```

Comment:
does `deepcopy` required here? `streams` dict can have millions or records and copying the entire dict recursively might cause performance problems. We don't need to make changes on cloned objects separately, also, we don't have `deepcopy` where filtering streams.

Reply:
You're right, actually we can even make `streams_to_flush` a list/set of stream names instead of dictionary given how only `streams_to_flush.keys()` are used.